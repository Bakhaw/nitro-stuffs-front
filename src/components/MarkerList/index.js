import React from 'react';
import PropTypes from 'prop-types';

import Marker from '../Marker';

function MarkerList({
  allVideos,
  isSelectedStuff,
  stuffShown,
  handleClickShowAllStuff,
  handleClickStuffSelected,
  handleClickVideoSelected,
}) {
  return (
    <div>
      <Marker
        additionalStyles={{ display: isSelectedStuff ? 'block' : 'none' }}
        x='90%'
        y='5%'
        height='35'
        width='35'
        onClick={() => handleClickShowAllStuff()}
        type='close'
      />
      {stuffShown.map((stuff) => {
        console.log({stuff})
        return (
          <Marker
            key={stuff.id}
            x={stuff.x - 17.5}
            y={stuff.y - 17.5}
            // x={`${(stuff.x * 100) / 750}%`}
            // y={`${(stuff.y * 100) / 720}%`}
            height='35'
            width='35'
            onClick={() => handleClickStuffSelected(stuff)}
            type={stuff.type}
          />
        )
      })}
      {allVideos.map((stuffSelected) => (
        <Marker
          key={stuffSelected.id}
          x={stuffSelected.x - 25}
          y={stuffSelected.y - 25}
          // x={`${(stuffSelected.x * 100) / 750}%`}
          // y={`${(stuffSelected.y * 100) / 720}%`}
          height='50'
          width='50'
          onClick={() => handleClickVideoSelected(stuffSelected)}
          type='position'
        />
      ))}
    </div>
  );
}

MarkerList.propTypes = {
  allVideos: PropTypes.array.isRequired,
  isSelectedStuff: PropTypes.bool.isRequired,
  stuffShown: PropTypes.array.isRequired,
  handleClickShowAllStuff: PropTypes.func.isRequired,
  handleClickStuffSelected: PropTypes.func.isRequired,
  handleClickVideoSelected: PropTypes.func.isRequired,
};

export default MarkerList;
