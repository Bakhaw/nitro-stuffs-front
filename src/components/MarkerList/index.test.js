import React from 'react';
import { shallow } from 'enzyme';

import MarkerList from '.';

test('should render properly', () => {
  const props = {
    allVideos: [
      {
        id: 'zokdaodq',
        title: 'Smoke CT from T Spawn',
        videoId: 'FqDYqI9o2YI',
        x: 254,
        y: 66,
      },
      {
        id: 'afdazdazzokdaodq',
        title: 'Smoke CT from Palace',
        videoId: 'FqDYqI9o2YI',
        x: 602,
        y: 352,
      },
    ],
    isSelectedStuff: false,
    stuffShown: [
      {
        id: '9avd46ed',
        type: 'flashbang',
        x: 157,
        y: 250,
        stuffPositions: [
          {
            id: 'zokdaodq',
            title: 'Smoke CT from T Spawn',
            videoId: 'FqDYqI9o2YI',
            x: 254,
            y: 66,
          },
          {
            id: 'afdazdazzokdaodq',
            title: 'Smoke CT from Palace',
            videoId: 'FqDYqI9o2YI',
            x: 602,
            y: 352,
          },
        ],
      },
    ],
    handleClickShowAllStuff: jest.fn(),
    handleClickStuffSelected: jest.fn(),
    handleClickVideoSelected: jest.fn(),
  };

  const container = shallow(<MarkerList {...props} />);
  expect(container.debug()).toMatchSnapshot();
});
