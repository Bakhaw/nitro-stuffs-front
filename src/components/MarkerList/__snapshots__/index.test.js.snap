// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
"<div>
  <Marker additionalStyles={{...}} x=\\"90%\\" y=\\"5%\\" height=\\"35\\" width=\\"35\\" onClick={[Function: onClick]} type=\\"close\\" />
  <Marker x=\\"20.933333333333334%\\" y=\\"34.72222222222222%\\" height=\\"35\\" width=\\"35\\" onClick={[Function: onClick]} type=\\"flashbang\\" />
  <Marker x=\\"33.86666666666667%\\" y=\\"9.166666666666666%\\" height=\\"50\\" width=\\"50\\" onClick={[Function: onClick]} type=\\"position\\" />
  <Marker x=\\"80.26666666666667%\\" y=\\"48.888888888888886%\\" height=\\"50\\" width=\\"50\\" onClick={[Function: onClick]} type=\\"position\\" />
</div>"
`;
